File: 0515-Find Largest Value in Each Tree Row/0515-Find Largest Value in Each Tree Row.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    vector<int> largestValues(TreeNode* root) {
        vector<int> result;
        if (root == nullptr)
        {
            return result;
        }
        queue<TreeNode*> Q;
        Q.push(root);
        while (!Q.empty())
        {
            int qs = Q.size();
            int maxVal = Q.front()->val;
            for (int i = 0; i < qs; ++i)
            {
                TreeNode* node = Q.front();
                Q.pop();
                maxVal = max(maxVal, node->val);
                if (node->left)
                {
                    Q.push(node->left);
                }
                
                if (node->right)
                {
                    Q.push(node->right);
                }
            }
            
            result.push_back(maxVal);
        }
        
        return result;
    }
};

File: 0501-0600/0515-Find Largest Value in Each Tree Row/0515-Find Largest Value in Each Tree Row.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

import collections
class Solution:
    def largestValues(self, root: TreeNode) -> List[int]:
        if root is None:
            return []
        Q = collections.deque([root])
        result = []
        while Q:
            qs = len(Q)
            maxVal = Q[0].val
            for _ in range(qs):
                node = Q.popleft()
                maxVal = max(maxVal, node.val)
                if node.left:
                    Q.append(node.left)
                if node.right:
                    Q.append(node.right)
            result.append(maxVal)
        return result
